src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java

```diff
@@ -163,26 +177,14 @@ public void onAtomicStep(@Nonnull StepAtomNode node, @Nonnull WorkflowRun run) {
             String principal = Objects.toString(node.getExecution().getAuthentication().getPrincipal(), "#null#");
             LOGGER.log(Level.FINE, () -> node.getDisplayFunctionName() + " - principal: " + principal);
 
-            SpanBuilder spanBuilder = null;
-            for (StepHandler stepHandler : ExtensionList.lookup(StepHandler.class)) {
-                if (stepHandler.canCreateSpanBuilder(node, run)) {
-                    try {
-                        spanBuilder = stepHandler.createSpanBuilder(node, run, getTracer());
-                    } catch (Exception e) {
-                        LOGGER.log(Level.WARNING, run.getFullDisplayName() + " failure to handle step " + node + " with handler " + stepHandler, e);
-                    }
-                    break;
-                }
-            }
-            if (spanBuilder == null) {
-                spanBuilder = getTracer().spanBuilder(node.getDisplayFunctionName());
-            }
+            StepHandler stepHandler = getStepHandlers().stream().filter(sh -> sh.canCreateSpanBuilder(node, run)).findFirst().get();
```

Comment:
I think it's a stepHandler rather than a step so "sh" makes sense to me. I can't use "stepHandler" as it would conflict with the already defined "stepHandler" variable.